src/github.com/getlantern/flashlight/flashlight.go

```diff
@@ -109,6 +111,34 @@ func runClientProxy(cfg *config.Config) {
 	// Configure client initially
 	client.Configure(cfg.Client)
 
+	// this flag specifies the port to open the HTTP server
+	// between the UI and
+	// with a corresponding HTTP server at
+	// the following address
+	if cfg.Client.UiAddr != "" {
+		wlChan := make(chan *whitelist.Config)
+		go func() {
+			http.ListenAndServe(cfg.Client, configUpdates, wlChan)
```

Comment:
Agreed. I'm returning an error now and also renamed the function to avoid confusion with ```ListenAndServe``` from ```net/http```